--- faac.bef/configure.in	2004-04-24 20:05:03.000000000 +0200
+++ faac.work/configure.in	2004-04-24 20:48:53.000000000 +0200
@@ -21,12 +21,23 @@
 AC_CHECK_DECL(strcasecmp, MY_DEFINE(HAVE_STRCASECMP))
 
 AC_CHECK_LIB(gnugetopt, getopt_long)
-AC_CHECK_LIB(mp4v2, MP4MetadataDelete,
-		    [AC_MSG_NOTICE([*** Building with MP4 support ***])
-		     MY_DEFINE([HAVE_LIBMP4V2])
-		     MP4LIBS="-lmp4v2"],
-		     [AC_MSG_NOTICE([*** Building without MP4 support ***])],
-		     -lstdc++)
+
+AC_ARG_WITH([mp4v2], [  --with-mp4v2	Build with MP4 support (default is yes.)],
+	    [ case "$withval" in
+	      yes|no) mp4v2="$withval" ;;
+	      *)  AC_MSG_ERROR([--with-mp4v2 does not take an argument]) ;;
+	      esac],
+	    [mp4v2=yes])
+
+if test "$mp4v2" = yes ; then
+	AC_CHECK_LIB(mp4v2, MP4MetadataDelete,
+		[AC_MSG_NOTICE([*** Building with MP4 support ***])
+		MY_DEFINE([HAVE_LIBMP4V2])
+		MP4LIBS="-lmp4v2"],
+		[AC_MSG_ERROR([*** Error - MP4v2 library not found - please, install MP4 library ***\n*** or use ./configure --without-mp4v2 ***])],
+		-lstdc++)
+fi
+
 AC_SUBST(MP4LIBS)
 AC_C_BIGENDIAN
 
